### PR TITLE
Only clean docs in development

### PR DESCRIPTION
**Pull request by Pezmc**
## Description

Replacement PR for https://github.com/flowforge/website/pull/966, which only cleans out the docs directly in development mode. 

`npm run start` (dev mode) will now:
 - Clean `site` (but not `img`)
 - Clear out `docs/**md`
 - Builds the javascript via terser 
 - Run `dev:docs` which copies across `../flowforge/docs`, if it exists
 - Run `dev:postcss`
 - Builds the website via netlify

In production, it'll remain as always, which Joe detailed in https://github.com/flowforge/website/pull/965#issuecomment-1633870796

## Related Issue(s)

#966 and #965

~Fixes 967~

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] I have considered the performance impact of these changes
 - [x] Suitable unit/system level tests have been added and they pass
 - [-] Documentation has been updated

### CODE REVIEW

**Comment by Pezmc** [comment]
@knolleary I had a working proof of concept for #967 ([464e613](https://github.com/flowforge/website/pull/969/commits/464e613242886a5810356949c658dd6d117f5192)) but have reverted it as it causes Netflify PR deploy previews to fail, as they _never_ have the docs/ only builds from the `live` branch; and I time boxed this to an hour.

File: package.json [hunk 1]
```diff
@@ -7,19 +7,19 @@
     "license": "Apache-2.0",
     "author": "Nick O'Leary",
     "scripts": {
-        "clean": "cross-env npx del-cli '_site/' && cross-env npx mkdirp '_site/js'",
-        "clean:dev": "cross-env npx del-cli '_site/!(img)' && cross-env npx mkdirp '_site/js'",
         "start": "npm-run-all clean:dev build:js --parallel dev:*",
-        "build": "cross-env NODE_ENV=production npm-run-all clean build:js --parallel prod:*",
+        "test": "mocha",
```
Comment (joepavitt) [inline package.json:11]: ???

Comment (joepavitt) [reply package.json:11]: why has this moved?

Comment (Pezmc) [reply package.json:11]: Previously in a seemingly random order, now alphabetical (and therefor grouped) except npm built in scripts start/test, see [c5501ad](https://github.com/flowforge/website/pull/969/commits/c5501ad7e29d495c556810b8f641f9d4b12e109d)

**Pull request merged**